mesonbuild/mdevenv.py

```diff
@@ -169,7 +169,7 @@ def run(options: argparse.Namespace) -> int:
         if options.dump is True:
             dump(devenv, varnames, dump_fmt)
         else:
-            with open(options.dump, "w", encoding='utf-8') as output:
+            with open(os.path.join(workdir, options.dump), "w", encoding='utf-8') as output:
```

Comment:
I think it should be `options.builddir` instead of `workdir`.

I'm a bit worried that this is not backward compatible, currently the path is relative to current directory and this makes it relative to builddir. That would break script doing:

```sh
meson setup builddir
meson devenv -C builddir --dump builddir/foo.env
```

OTOH, I'm sure @eli-schwartz would be happy to point out that `-C` means `chdir()` which kind of implies that paths should be relative to builddir...

In any case, currently the behaviour of relative path is undocumented, so we should document it, and if we change current behaviour it should also have release notes.

Reply:
Yeah, I think that makes sense both semantically and in terms of what people actually want.

e.g. I doubt the dump file is particularly useful in the source directory, right? It refers to files in the build directory anyway.

Reply:
or maybe this filename  should be automatically generated which will avoid errors. So remove the dump options and name the result `meson-xxx.env`

Reply:
Following the discussion in https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/3719#note_1869567, I'd prefer to configure with a `--devenv-format=vscode` (maybe a better name) which will maintain up2date a file in the buildfolder such as meson-vscode.env in addition to the regular devenv. I dont think we should generate always all the files. 
It should be generated on demand at (re)configure time.

Reply:
The number of `meson setup` options we have already is a big cognitive burden, I don't think that this is useful enough to add another one.

Please find some way that doesn't include adding a new option to the `meson setup` command.

Reply:
I agree but gcc or cmake have also a lot of CLI options. 

I was wondering also if we could extend the `--vsenv` as it could be a bit confusing. Why not deprecate `--vsenv` progressively  and enable `--devenv`.